rustls/src/crypto/signer.rs

```diff
@@ -12,6 +12,21 @@ use core::fmt::Debug;
 ///
 /// This interface is used by rustls to use a private signing key
 /// for authentication.  This includes server and client authentication.
+///
+/// Objects of this type are always and only provided by
```

Comment:
You can actually just make these outside of `load_private_key()`, eg, see https://docs.rs/rustls/0.22.0-alpha.4/rustls/crypto/ring/sign/fn.any_ecdsa_type.html and co. You would do this if you wanted to minimise the amount of stuff linked into your binary, or wanted to have a single server with multiple sets of keys and certificates (via, https://docs.rs/rustls/0.22.0-alpha.4/rustls/server/struct.ResolvesServerCertUsingSni.html or some other implementation of https://docs.rs/rustls/0.22.0-alpha.4/rustls/server/trait.ResolvesServerCert.html)

Reply:
Which leads to the question: what's the point of `load_private_key()`? It's to regularise the dependency from the `ConfigBuilder` states on the crypto provider.

Reply:
Aha, thanks for the pointers. Now I see more of the picture. Will update.